client/api/stripe/service.ts

```
@@ -21,4 +26,85 @@ const connectStripe = async (formData: object) => {
   }
 };
 
-export { connectStripe };
+export const createPaymentIntent = async (
+  stripeAccountId: string,
+  form_id: string,
+  userId: string,
+  price: number,
+  fee: number,
+): Promise<PaymentIntent | null> => {
+  try {
+    const response = await fetch(
+      `/api/accounts/${stripeAccountId}/paymentIntent`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          price,
+          fee,
+          form_id,
+          userId,
+        }),
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error(`Error creating payment intent: ${response.status}`);
+    }
+
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error("Error creating payment intent:", error);
+    return null;
+  }
+};
+
+export const getPublishableKey = async (): Promise<string> => {
+  try {
+    const response = await fetch("/api/accounts/key/publishable");
+
+    if (!response.ok) {
```

nit: seems like a different error here. We are catching the error but if we receive a non `ok` response status, maybe throw a different error

---

ack!